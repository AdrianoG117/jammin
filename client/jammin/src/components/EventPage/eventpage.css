.event-container {
  background-color: rgba(0, 0, 0, 0.5);
  width: 90%;
  height: 85%;
  margin: 2rem auto;
  color: white;
  display: flex;
}

.event-data {
  /* background-color: green; */
  width: 65%;
  display: flex;
  flex-direction: column;
}

.data-item {
  /* margin: 1rem */
  padding: 1rem;
  display: flex;
  position: relative;
}

#participants {
  margin-left: auto;
  position: relative;
  right: 35px;
  bottom: 27px;
}

.event-added-btn {
  margin-left: auto;
  background-color: white;
  padding: 1rem;
  border-radius: 10px;
}

.add-btn {
  margin-left: auto;
  background-color: white;
  padding: 1rem;
  border-radius: 10px;
}

.add-btn:hover {
  background-color: rgba(0, 0, 0, 0.5);
  color: white;
  cursor: pointer;
}

#date {
  font-size: 2rem;
}

#title {
  font-size: 3rem;
  font-weight: bold;
}
#location {
}
#description {
  margin: 2rem 4rem;
  font-size: 1.5rem;
}

#languages {
  border-bottom: 1px solid white;
  padding-bottom: 2rem;
}

.socialmap {
  display: flex;
  flex-direction: column;
  width: 35%;
  border: 1px solid white;
}

.map-container {
  height: 35%;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid white;
}

.pin-img {
  height: 5vh;
  color: white;
  justify-self: self;
}

.img-container {
  display: flex;
  justify-content: center;
  align-items: center;
}

.img-container img {
  margin: 0.6rem;
}

.img-container h2 {
}
