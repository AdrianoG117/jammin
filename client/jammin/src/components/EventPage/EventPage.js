import React, { useState, useEffect } from 'react';
import './eventpage.css';
import { useHistory } from 'react-router-dom';
import Social from '../Social/Social.jsx';
import { GoogleMap, useLoadScript, Marker } from '@react-google-maps/api';
import apiService from '../../ApiService';
import Pin from '../../images/placeholder.png';
import Voice from '../../images/voice.png';
import moment from 'moment';

const initialState = {
  name: '',
  message: '',
};

function EventPage(props) {
  const pathname = window.location.pathname;
  const urlID = pathname.slice(6);
  //initial state => from props || create another object for  initial
  const [data, setData] = useState(props.location?.state?.jam);
  const [msg, setMsg] = useState(initialState); //message state
  const userData = props.userData;
  const setUserData = props.setUserData;
  const isSignedUp = props.isSignedUp;
  const setIsSignedUp = props.isSignedUp;

  // const [num, setNum] = useState(data.numOfParticipants);

  const history = useHistory();

  //create another state with initial

  useEffect(() => {
    apiService.getEvent(urlID).then((data) => {
      setData(data[0]);
    });
  }, [msg]);

  //BUG TO SOLVE WHEN PUTTING DATA WITH MESSAGES

  // useEffect(() => {
  //   apiService.getEvent(urlID).then((data) => {
  //     setData(data[0]);
  //   });
  // }, [data]);

  // const data = props.location.state.jam;
  const libraries = ['places'];
  const mapContainerStyle = {
    width: '100%',
    height: '96%',
    borderRadius: '10px',
    marginBottom: '10px',
  };

  const { isLoaded, loadError } = useLoadScript({
    googleMapsApiKey: 'AIzaSyCaWssSgkyqO9SyAJ7VvTonQ1ASzdyQ6oM',
    libraries,
  });

  if (loadError) return 'Error loading maps';
  if (!isLoaded) return 'Loading Maps';

  const center = data?.locCords;

  async function addToEvents(userid, jamid) {
    const body = {
      id: userid,
      jamId: jamid,
    };

    const idToSend = {
      id: jamid,
    };
    await apiService.addjam(body);
    console.log(jamid);
    await apiService.addParticipant(idToSend);
    //send back from post request
    setUserData((previous) => ({
      ...previous,
      comingEvents: [...previous.comingEvents, data],
    }));
    setData((previous) => {
      console.log('previous', previous);
      return {
        ...previous,
        numOfParticipants: previous.numOfParticipants + 1,
      };
    });
    // setNum((previous) => previous + 1);
  }

  function isEventAdded(jamid) {
    const arr = userData.comingEvents;
    for (let i = 0; i < arr.length; i++) {
      if (arr[i]._id === jamid) {
        return true;
      }
    }
    return false;
  }

  return (
    <div className="event-container">
      {data && (
        <>
          <div className="event-data">
            <div className="data-item" id="date">
              <h2>{moment(data.date).format('MMM Do, h:mm a')}</h2>
              {isSignedUp ? (
                isEventAdded(data._id) ? (
                  <button className="event-added-btn">EVENT ADDED</button>
                ) : (
                  <button
                    className="add-btn"
                    onClick={() => addToEvents(userData._id, data._id)}
                  >
                    PARTICIPATE
                  </button>
                )
              ) : (
                <button
                  className="add-btn"
                  onClick={() => {
                    history.push('/login');
                  }}
                >
                  Login to participate
                </button>
              )}
            </div>
            <div className="data-item">
              <h1 id="title">{data.title}</h1>
              <p id="participants">{data.numOfParticipants} going</p>
            </div>
            <div className="data-item" id="location">
              <div className="img-container">
                <img className="pin-img" src={Pin} alt="" />
                <h2>{data.location}</h2>
              </div>
            </div>
            <div className="language-container">
              <div className="voice-icon-container">
                <img src={Voice} id="voice" alt="" />
              </div>
              <h2 className="data-item" id="languages">
                {data.languages}
              </h2>
            </div>
            <p className="data-item" id="description">
              {data.description}
            </p>
          </div>
          <div className="socialmap">
            <div className="map-container">
              <GoogleMap
                mapContainerStyle={mapContainerStyle}
                zoom={15}
                center={center}
              >
                <Marker position={center} />
              </GoogleMap>
            </div>
            <Social
              jam={data}
              msg={msg}
              setMsg={setMsg}
              initialState={initialState}
              isSignedUp={isSignedUp}
              userData={userData}
            />
          </div>
        </>
      )}
    </div>
  );
}

export default EventPage;
