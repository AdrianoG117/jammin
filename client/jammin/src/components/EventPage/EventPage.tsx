import React, { useState, useEffect, FunctionComponent } from "react";
import "./eventpage.css";
import { useHistory } from "react-router-dom";
import Social from "../Social/Social";
import { GoogleMap, useLoadScript, Marker } from "@react-google-maps/api";
import apiService from "../../ApiService";
import Pin from "../../images/placeholder.png";
import Voice from "../../images/voice.png";
import moment from "moment";

const initialState = {
  name: "",
  message: "",
};

const apiKey = process.env.REACT_APP_GOOGLE_MAPS_API_KEY;

function EventPage (props): FunctionComponent {
  const pathname = window.location.pathname;
  const urlID = pathname.slice(6);
  const [data, setData] = useState(props.location?.state?.jam);
  const [msg, setMsg] = useState(initialState); //message state
  const userData = props.userData;
  const setUserData = props.setUserData;
  const isSignedUp = props.isSignedUp;

  const history = useHistory();

  useEffect(() => {
    apiService.getEvent(urlID).then((data) => {
      setData(data[0]);
    });
  }, [msg]);

  const libraries = ["places"];
  const mapContainerStyle = {
    width: "100%",
    height: "96%",
    borderRadius: "10px",
    marginBottom: "10px",
  };

  const { isLoaded, loadError } = useLoadScript({
    googleMapsApiKey: apiKey,
    libraries,
  });

  if (loadError) return "Error loading maps";
  if (!isLoaded) return "Loading Maps";

  const center = data?.locCords;
  // on click used to increment the participants going(number of participants)
  async function addToEvents(userid, jamid) {
    const body = {
      id: userid,
      jamId: jamid,
    };

    const idToSend = {
      id: jamid,
    };
    await apiService.addjam(body);
    await apiService.addParticipant(idToSend);
    setUserData((previous) => ({
      ...previous,
      comingEvents: [...previous.comingEvents, data],
    }));
    setData((previous) => {
      return {
        ...previous,
        numOfParticipants: previous.numOfParticipants + 1,
      };
    });
  }

  function isEventAdded(jamid) {
    const arr = userData.comingEvents;
    console.log("arrrr", arr);
    for (let i = 0; i < arr.length; i++) {
      if (arr[i]._id === jamid) {
        return true;
      }
    }
    return false;
  }

  return (
    <div className="event-container">
      {data && (
        <>
          <div className="event-data">
            <div className="data-item" id="date">
              <h2>{moment(data.date).format("MMM Do, h:mm a")}</h2>
              {isSignedUp ? (
                isEventAdded(data._id) ? (
                  <button className="event-added-btn">EVENT ADDED</button>
                ) : (
                  <button
                    className="add-btn"
                    onClick={() => addToEvents(userData._id, data._id)}
                  >
                    PARTICIPATE
                  </button>
                )
              ) : (
                <button
                  className="add-btn"
                  onClick={() => {
                    history.push("/login");
                  }}
                >
                  Login to participate
                </button>
              )}
            </div>
            <div className="data-item">
              <h1 id="title">{data.title}</h1>
              <p id="participants">{data.numOfParticipants} going</p>
            </div>
            <div className="data-item" id="location">
              <div className="img-container">
                <img className="pin-img" src={Pin} alt="" />
                <h2>{data.location}</h2>
              </div>
            </div>
            <div className="language-container">
              <div className="voice-icon-container">
                <img src={Voice} id="voice" alt="" />
              </div>
              <h2 className="data-item" id="languages">
                {data.languages}
              </h2>
            </div>
            <p className="data-item" id="description">
              {data.description}
            </p>
          </div>
          <div className="socialmap">
            <div className="map-container">
              <GoogleMap
                mapContainerStyle={mapContainerStyle}
                zoom={15}
                center={center}
              >
                <Marker position={center} />
              </GoogleMap>
            </div>
            <Social
              jam={data}
              msg={msg}
              setMsg={setMsg}
              initialState={initialState}
              isSignedUp={isSignedUp}
              userData={userData}
            />
          </div>
        </>
      )}
    </div>
  );
}

export default EventPage;

