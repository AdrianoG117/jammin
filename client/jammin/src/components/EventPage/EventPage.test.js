import {render, screen} from '@testing-library/react';
import EventPage from './EventPage';

const mockProps = {
  userData: null,
  setUserData: () =>{},
  isSignedUp: false,
  setIsSignedUp: ()=>{},
  
  isEventAdded: (jamid) => {
    const arr = [{_id:"615aa8eb25df7e7d9ed72fbc"}];
    for (let i = 0; i < arr.length; i++) {
      if (arr[i]._id === jamid) {
        return true;
      }
    }
    return false;
  }
}

describe('EventPage component', ()=> {
  
  test('should match the snapshot', () => {
    const {container} = render(<EventPage  
     userData={mockProps.userData}
                  setUserData={mockProps.setUserData}
                  isSignedUp={mockProps.isSignedUp}
                  setIsSignedUp={mockProps.setIsSignedUp}
    />);
    expect(container.firstChild).toMatchSnapshot();
  })
  
   test('is EventAdded function should return true or false if the event is added or not added', () => {
     expect(mockProps.isEventAdded("615aa8eb25df7e7d9ed72fbc")).toBeTruthy();
    expect( mockProps.isEventAdded("")).toBeFalsy()     
   })
   
  
  test('Should render the headings when user is signed up', () => {
    render (<EventPage  
     userData={mockProps.userData}
                  setUserData={mockProps.setUserData}
                  isSignedUp={mockProps.isSignedUp}
                  setIsSignedUp={mockProps.setIsSignedUp}
    />);
    //if user signed up, then the paritciapte button should render
    //else the pariticpate button should be rendered
    
    //If user is logged in, then check if user has the event added
    //if true then it should show EVENT ADDED else  show PARTICIPATE
    if(mockProps.isSignedUp){
      if(mockProps.isEventAdded('615aa8eb25df7e7d9ed72fbc')){
        screen.getByText(/EVENT ADDED/)   
      }
      if( !mockProps.isEventAdded('')){
        screen.getByText(/PARTICIPATE/)   
      }
      
    }
  })
  
  
  
  test('Should render the headings when user is NOT signed up', () => {
    render (<EventPage  
     userData={mockProps.userData}
                  setUserData={mockProps.setUserData}
                  isSignedUp={mockProps.isSignedUp}
                  setIsSignedUp={mockProps.setIsSignedUp}
    />);
    
    if(mockProps.isSignedUp){
     screen.getByText(/Login to participate/)   
      
    }
  })

})