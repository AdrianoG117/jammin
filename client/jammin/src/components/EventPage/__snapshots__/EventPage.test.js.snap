// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventPage component should match the snapshot 1`] = `Loading Maps`;
